### PR TITLE
Some more builtins

### LINKED ISSUE #98: cljs unit test failures don't seem to fail the build
One test seem to be failing in the CLJS unit tests, but the builds are still passing. We should fix this.

https://github.com/johanfylling/jarl/runs/7510168600?check_suite_focus=true

Screenshot 2022-07-26 at 00 52 46

### PR DESCRIPTION
**Pull request by anderseknert**
A few new built-in function implementations, and some various other fixes. Note that these commits are rather separate, so probably better to merge them as they are presented rather than squashing them into one.

Fixes #98

### CODE REVIEW

File: src/main/cljc/jarl/builtins/graphs.cljc [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+          (swap! visited conj node)))
+      (fn children [node]
+        (get graph node))
+      root)))
+
+(defn builtin-graph-reachable
+  [{[graph initial] :args}]
+  (if (empty? graph)
+    #{}
+    (set (mapcat (partial graph-seq graph) initial))))
```
Comment (johanfylling) [inline src/main/cljc/jarl/builtins/graphs.cljc:19]: `partial` for the win! :)

File: src/main/cljc/vendor/semver/core.cljc [hunk 2]
```diff
@@ -0,0 +1,72 @@
+; Courtesy of https://github.com/owainlewis/semver
```
Comment (johanfylling) [inline src/main/cljc/vendor/semver/core.cljc:1]: This isn't published to Clojars? If not, there aren't any published alternatives up to snuff?

File: src/main/cljc/jarl/eval.cljc [hunk 3]
```diff
@@ -474,10 +474,16 @@
       (catch ExceptionInfo e
         (let [type (errors/ex-type e)]
           (if (or (= type :jarl.exceptions/builtin-exception) (= type :jarl.exceptions/type-exception))
-            (if (= name "regex.is_valid")
+            (condp = name
```
Comment (johanfylling) [inline src/main/cljc/jarl/eval.cljc:477]: Perhaps we could add the expected return value as an attribute on the exception 🤔 . That way, we can keep this per-builtin special knowledge closer to the implementation.

**Review by johanfylling — APPROVED** [review]
LGTM 👍 
Would be good if we could solve the semver dependency some other way than copy/paste, though.

File: src/main/cljc/jarl/builtins/graphs.cljc (hunk 1 above)
Comment (anderseknert) [reply src/main/cljc/jarl/builtins/graphs.cljc:19]: Indeed! Just so nice 👌

File: src/main/cljc/vendor/semver/core.cljc (hunk 2 above)
Comment (anderseknert) [reply src/main/cljc/vendor/semver/core.cljc:1]: An older version is: https://clojars.org/search?q=semver

That it's older doesn't matter, but it has a few calls to `Integer/parseInt` in it, which makes it Java only. The other ones had the same problems AFAICS. I could try submitting a PR to the repo, but he seemed pretty inactive on Github in recent years, so this seemed like the simpler solution for the moment.

File: src/main/cljc/jarl/eval.cljc (hunk 3 above)
Comment (anderseknert) [reply src/main/cljc/jarl/eval.cljc:477]: Yeah. On the other hand, I kinda _want_ these exceptions to be annoying, as it increases the chances of these things getting fixed in OPA. The graph reachable ones looks like they can be removed from here not too long from now 😍 https://github.com/open-policy-agent/opa/pull/4956

**Pull request merged**